data-service/src/test/java/com/amazonaws/blox/dataservice/api/DescribeEnvironmentApiTest.java

```diff
@@ -0,0 +1,142 @@
+            .activeEnvironmentRevisionId(ACTIVE_ENVIRONMENT_REVISION_ID)
+            .latestEnvironmentRevisionId(ACTIVE_ENVIRONMENT_REVISION_ID)
+            .deploymentConfiguration(new DeploymentConfiguration())
+            .deploymentMethod(DEPLOYMENT_METHOD)
+            .validEnvironment(true)
+            .build();
+    describeEnvironmentRequest =
+        DescribeEnvironmentRequest.builder().environmentId(environmentIdWrapper).build();
+  }
+
+  @Test
+  public void describeEnvironmentSuccess() throws Exception {
+    when(environmentRepository.getEnvironment(isA(EnvironmentId.class))).thenReturn(environment);
+    final DescribeEnvironmentResponse describeEnvironmentResponse =
+        describeEnvironmentApi.describeEnvironment(describeEnvironmentRequest);
+
+    verify(environmentRepository).getEnvironment(environmentIdArgumentCaptor.capture());
+    assertEquals(environmentIdArgumentCaptor.getValue(), environmentId);
+
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentId(), environmentIdWrapper);
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentHealth(),
+        environment.getEnvironmentHealth().name());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getActiveEnvironmentRevisionId(),
+        environment.getActiveEnvironmentRevisionId());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getCreatedTime(),
+        environment.getCreatedTime());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentStatus(),
+        environment.getEnvironmentStatus().name());
+    assertEquals(describeEnvironmentResponse.getEnvironment().getRole(), environment.getRole());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getDeploymentConfiguration(),
+        apiModelMapper.toWrapperDeploymentConfiguration(environment.getDeploymentConfiguration()));
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentType().name(),
+        environment.getEnvironmentType().name());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getLastUpdatedTime(),
+        environment.getLastUpdatedTime());
+  }
+
+  @Test(expected = ResourceNotFoundException.class)
+  public void describeEnvironmentResourceNotFoundException() throws Exception {
+    when(environmentRepository.getEnvironment(isA(EnvironmentId.class)))
+        .thenThrow(new ResourceNotFoundException(ResourceType.ENVIRONMENT, ENVIRONMENT_NAME));
+    describeEnvironmentApi.describeEnvironment(describeEnvironmentRequest);
+  }
+
+  @Test(expected = InternalServiceException.class)
+  public void describeEnvironmentInternalServiceException() throws Exception {
+    when(environmentRepository.getEnvironment(isA(EnvironmentId.class)))
+        .thenThrow(new InternalServiceException(""));
+    describeEnvironmentApi.describeEnvironment(describeEnvironmentRequest);
+  }
+}
```

Comment:
Should add one more test case to cover catching unknown exception and wrap to InternalServiceException.

Reply:
Good point. Done.